internal/resources/policy/label_selector_requirement_schema.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright © 2022 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package policy
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/helper"
+	policymodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/policy"
+)
+
+var labelSelectorRequirement = &schema.Resource{
+	Description: "Metadata describing the type of the resource",
+	Schema: map[string]*schema.Schema{
+		KeyKey: {
+			Type:        schema.TypeString,
+			Description: "Key is the label key that the selector applies to",
+			Optional:    true,
+		},
+		OperatorKey: {
+			Type:         schema.TypeString,
+			Description:  "Operator represents a key's relationship to a set of values",
+			Optional:     true,
+			ValidateFunc: validation.StringInSlice([]string{"In", "NotIn", "Exists", "DoesNotExist"}, false),
+		},
+		ValuesKey: {
+			Type:        schema.TypeList,
+			Description: "Values is an array of string values",
+			Required:    true,
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
+		},
+	},
+}
+
+func constructLabelSelectorRequirement(data interface{}) (labelSelectorRequirement *policymodel.K8sIoApimachineryPkgApisMetaV1LabelSelectorRequirement) {
+	if data == nil {
+		return labelSelectorRequirement
+	}
+
+	labelSelectorRequirementData, _ := data.(map[string]interface{})
+
+	labelSelectorRequirement = &policymodel.K8sIoApimachineryPkgApisMetaV1LabelSelectorRequirement{
+		Values: make([]string, 0),
+	}
+
+	if v, ok := labelSelectorRequirementData[KeyKey]; ok {
+		helper.SetPrimitiveValue(v, &labelSelectorRequirement.Key, KeyKey)
+	}
+
+	if v, ok := labelSelectorRequirementData[OperatorKey]; ok {
+		helper.SetPrimitiveValue(v, &labelSelectorRequirement.Operator, OperatorKey)
+	}
+
+	if v, ok := labelSelectorRequirementData[ValuesKey]; ok {
+		vs, _ := v.([]interface{})
+		for _, raw := range vs {
+			labelSelectorRequirement.Values = append(labelSelectorRequirement.Values, raw.(string))
```

Comment:
will raw.(string) cause runtime panic ?

Reply:
No, because the values have to be string, that is ensured in the schema constraints where it is defined as a type string. 

The plan itself would throw an error before any call to this construct happens in case the values are not string.